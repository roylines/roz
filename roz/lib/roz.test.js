const {handler} = require('./roz');

const {error} = require('lambda-log');
const telegramHandler = require('./handlers/telegram');
const pollHandler = require('./handlers/poll');
const directHandler = require('./handlers/direct');

jest.mock('./handlers/telegram');
jest.mock('./handlers/poll');
jest.mock('./handlers/direct');
jest.mock('lambda-log');

describe('roz', () => {
  beforeEach(() => jest.resetAllMocks());
  describe('telegram message', () => {
    it('should return 200 if handler throws', async () => {
      telegramHandler.handle.mockRejectedValue('FAILED TELEGRAM');
      const response = await handler({body: 'BODY'}, 'CONTEXT');
      expect(response).toMatchSnapshot({statusCode: 200});
      expect(error).toMatchSnapshot();
    });
    it('should call handler correctly', async () => {
      telegramHandler.handle.mockResolvedValue('TELEGRAM');
      const response = await handler({body: 'BODY'}, 'CONTEXT');
      expect(response).toMatchSnapshot('TELEGRAM');
    });
  });

  describe('poll', () => {
    it('should return 200 if handler throws', async () => {
      pollHandler.handle.mockRejectedValue('FAILED POLL');
      const response = await handler(
        {'detail-type': 'Scheduled Event'},
        'CONTEXT',
      );
      expect(response).toMatchSnapshot({statusCode: 200});
      expect(error).toMatchSnapshot();
    });
    it('should call handler correctly', async () => {
      pollHandler.handle.mockResolvedValue('POLL');
      const response = await handler(
        {'detail-type': 'Scheduled Event'},
        'CONTEXT',
      );
      expect(response).toMatchSnapshot('POLL');
    });
  });

  describe('direct', () => {
    it('should return 200 if handler throws', async () => {
      directHandler.handle.mockRejectedValue('FAILED DIRECT');
      const response = await handler({}, 'CONTEXT');
      expect(response).toMatchSnapshot({statusCode: 200});
      expect(error).toMatchSnapshot();
    });
    it('should call handler correctly', async () => {
      directHandler.handle.mockResolvedValue('DIRECT');
      const response = await handler({}, 'CONTEXT');
      expect(response).toMatchSnapshot('DIRECT');
    });
  });
});
