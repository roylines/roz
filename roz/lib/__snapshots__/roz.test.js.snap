// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`roz direct should call handler correctly: DIRECT 1`] = `"DIRECT"`;

exports[`roz direct should return 500 if handler throws 1`] = `
Object {
  "statusCode": 500,
}
`;

exports[`roz direct should return 500 if handler throws 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "FAILED DIRECT",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`roz poll should call handler correctly: POLL 1`] = `"POLL"`;

exports[`roz poll should return 500 if handler throws 1`] = `
Object {
  "statusCode": 500,
}
`;

exports[`roz poll should return 500 if handler throws 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "FAILED POLL",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`roz telegram message should call handler correctly: TELEGRAM 1`] = `"TELEGRAM"`;

exports[`roz telegram message should return 500 if handler throws 1`] = `
Object {
  "statusCode": 500,
}
`;

exports[`roz telegram message should return 500 if handler throws 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "FAILED TELEGRAM",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
